apps/ui/app/pages/[schema]/pages/[page]/edit.vue

```diff
@@ -0,0 +1,218 @@
+      variables: { page: pageData.value },
+    },
+  }).catch((err) => {
+    statusModalData.value = {
+      type: "error",
+      message: err.error ? err.error[0].message : err,
+    };
+    console.log(statusModalData.value.message);
+  });
+
+  if (response.data?.save) {
+    statusModalData.value = {
+      type: "success",
+      message: `Updated "${page}"`,
+    };
+  }
+  showStatusModal.value = true;
+}
+
+const enableSaveButton = computed<boolean>(
+  () => hasUnsavedHtml.value || hasUnsavedCss.value || hasUnsavedJs.value
+);
+
+function hasUnsavedChanges(
+  type: string,
+  newValue: string,
+  oldValue: string
+): boolean {
+  const newValueStr = JSON.stringify(newValue);
+  const oldValueStr = JSON.stringify(oldValue);
+  const originalValueStr = JSON.stringify(
+    originalPageData.value[type as keyof DeveloperPage]
+  );
+  return newValueStr !== oldValueStr && newValueStr !== originalValueStr;
+}
+
+watch(
+  () => pageData.value.html,
+  (newValue, oldValue) => {
+    hasUnsavedHtml.value = hasUnsavedChanges(
+      "html",
+      newValue as string,
+      oldValue as string
+    );
+  }
+);
+
+watch(
+  () => pageData.value.css,
+  (newValue, oldValue) => {
+    hasUnsavedCss.value = hasUnsavedChanges(
+      "css",
+      newValue as string,
+      oldValue as string
+    );
+  }
+);
+
+watch(
+  () => pageData.value.javascript,
+  (newValue, oldValue) => {
+    hasUnsavedJs.value = hasUnsavedChanges(
+      "javascript",
+      newValue as string,
+      oldValue as string
+    );
+  }
+);
+
+const crumbs: Record<string, string> = {};
+crumbs[schema as string] = `/${schema}`;
+crumbs["Pages"] = `/${schema}/pages`;
+crumbs[page as string] = `/${schema}/pages/${page}`;
+crumbs["Edit"] = "";
+</script>
+
+<template>
+  <Container class="relative">
+    <PageHeader :title="`Edit ${page}`" align="left">
+      <template #prefix>
+        <BreadCrumbs :crumbs="crumbs" align="left" />
+      </template>
+    </PageHeader>
+    <div v-if="!isAdmin">
+      <Message id="page-editor-error" :invalid="true">
+        <p>Please sign in as "admin" to modify page content.</p>
+      </Message>
+    </div>
+    <template v-else>
+      <div
+        class="sticky top-0 w-full flex justify-end items-center bg-navigation py-2 gap-5 px-7.5 z-10 mb-5"
+      >
+        <NuxtLink
+          :to="`/${schema}/pages/${page}`"
+          class="font-display tracking-widest uppercase text-button-text text-heading-lg hover:underline"
+        >
+          View page
+        </NuxtLink>
+        <Button
+          type="primary"
+          size="small"
+          :disabled="!enableSaveButton"
+          :class="{
+            'animate-pulse': enableSaveButton,
+            'cursor-not-allowed': !enableSaveButton,
+          }"
+          @click="saveSetting"
+        >
+          Save Changes
+        </Button>
+      </div>
+      <div class="grid grid-cols-2 gap-7.5">
```

Comment:
should this div be a 'layout' component?
(I am still trying to get us to create layout components so we can ensure consistent layout changes, but maybe that is not feasible)